### PR TITLE
Add regression test for AMR w/ high-order SCALARs

### LINKED ISSUE #12309: Crash with adaptivity + high order scalar variables
## Rationale

Moose crashes (or bails out with an assertion) if the mesh gets refined in a simulation that has <s>stateful material properties and</s> high order scalar variables (I'm assuming it's the scalar vars but I'll be continuing to check this).

I might me in uncharted territory here (or assuming I can use a capability that simply doesn't exist yet). But I'd expect a verbose error message rather than a crash nonetheless.

## Description

Still working on a minimal input that works with vanilla MOOSE, but with my Marmot branch I get:

```
Assertion `var < this->n_vars(s)' failed.
var = 13
this->n_vars(s) = 13


Stack frames: 20
0: 0   libmesh_dbg.0.dylib                 0x000000010c3dc050 libMesh::print_trace(std::__1::basic_ostream<char, std::__1::char_traits<char> >&) + 2192
1: 1   libmesh_dbg.0.dylib                 0x000000010c3c62b1 libMesh::MacroFunctions::report_error(char const*, int, char const*, char const*) + 369
2: 2   libmagpie-dbg.0.dylib               0x000000010666483f libMesh::DofObject::n_comp(unsigned int, unsigned int) const + 3551
3: 3   libmesh_dbg.0.dylib                 0x000000010def89f0 libMesh::OldSolutionValue<double, &(void libMesh::FEMContext::point_value<double>(unsigned int, libMesh::Point const&, double&, double) const)>::eval_at_node(libMesh::FEMContext const&, unsigned int, unsigned int, libMesh::Node const&, double) + 144
4: 4   libmesh_dbg.0.dylib                 0x000000010deeb99d libMesh::GenericProjector<libMesh::OldSolutionValue<double, &(void libMesh::FEMContext::point_value<double>(unsigned int, libMesh::Point const&, double&, double) const)>, libMesh::OldSolutionValue<libMesh::VectorValue<double>, &(void libMesh::FEMContext::point_gradient<libMesh::VectorValue<double> >(unsigned int, libMesh::Point const&, double&, double) const)>, double, libMesh::VectorSetAction<double> >::operator()(libMesh::StoredRange<libMesh::MeshBase::const_element_iterator, libMesh::Elem const*> const&) const + 14221
5: 5   libmesh_dbg.0.dylib                 0x000000010decc50c void libMesh::Threads::parallel_for<libMesh::StoredRange<libMesh::MeshBase::const_element_iterator, libMesh::Elem const*>, libMesh::GenericProjector<libMesh::OldSolutionValue<double, &(void libMesh::FEMContext::point_value<double>(unsigned int, libMesh::Point const&, double&, double) const)>, libMesh::OldSolutionValue<libMesh::VectorValue<double>, &(void libMesh::FEMContext::point_gradient<libMesh::VectorValue<double> >(unsigned int, libMesh::Point const&, double&, double) const)>, double, libMesh::VectorSetAction<double> > >(double const&, libMesh::GenericProjector<libMesh::OldSolutionValue<double, &(void libMesh::FEMContext::point_value<double>(unsigned int, libMesh::Point const&, double&, double) const)>, libMesh::OldSolutionValue<libMesh::VectorValue<double>, &(void libMesh::FEMContext::point_gradient<libMesh::VectorValue<double> >(unsigned int, libMesh::Point const&, double&, double) const)>, double, libMesh::VectorSetAction<double> > const&) + 124
6: 6   libmesh_dbg.0.dylib                 0x000000010deca512 libMesh::System::project_vector(libMesh::NumericVector<double> const&, libMesh::NumericVector<double>&, int) const + 10594
7: 7   libmesh_dbg.0.dylib                 0x000000010dec79c9 libMesh::System::project_vector(libMesh::NumericVector<double>&, int) const + 153
8: 8   libmesh_dbg.0.dylib                 0x000000010ddee431 libMesh::System::restrict_vectors() + 865
9: 9   libmesh_dbg.0.dylib                 0x000000010ddef2c9 libMesh::System::prolong_vectors() + 25
10: 10  libmesh_dbg.0.dylib                 0x000000010dd2b3d5 libMesh::EquationSystems::reinit_solutions() + 10293
11: 11  libmoose-dbg.0.dylib                0x000000010952852b FEProblemBase::meshChangedHelper(bool) + 395
12: 12  libmoose-dbg.0.dylib                0x00000001095281fa FEProblemBase::adaptMesh() + 698
13: 13  libmoose-dbg.0.dylib                0x000000010901b2cd Transient::incrementStepOrReject() + 125
14: 14  libmoose-dbg.0.dylib                0x000000010901b067 Transient::execute() + 119
15: 15  libmoose-dbg.0.dylib                0x000000010a63c30b MooseApp::executeExecutioner() + 235
16: 16  libmoose-dbg.0.dylib                0x000000010a63cc5f MooseApp::run() + 335
17: 17  marmot-dbg                          0x0000000105adc396 main + 278
18: 18  libdyld.dylib                       0x00007fff74dbd085 start + 1
19: 19  ???                                 0x0000000000000003 0x0 + 3
[0] /Users/schwd/Programs/libmesh/installed/include/libmesh/dof_object.h, line 807, compiled nodate at notime
application called MPI_Abort(MPI_COMM_WORLD, 1) - process 0
[unset]: write_line error; fd=-1 buf=:cmd=abort exitcode=1
:
system msg for write_line failure : Bad file descriptor
```

(this error is triggered in serial _and_ parallel runs)

### PR DESCRIPTION
**Pull request by roystgnr**
This resolves #12309 and #12538

The libMesh bug here got fixed years ago, but apparently the MOOSE test coverage for the fix got lost in the couch cushions until I stumbled across it while searching old issues.

### CODE REVIEW

File: test/tests/adaptivity/scalar/tests [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[Tests]
+  issues = '#12538 #12309'
+  design = 'syntax/Adaptivity/index.md'
+  [./test]
+    type = 'Exodiff'
+    input = 'scalar_adaptivity.i'
+    exodiff = 'scalar_adaptivity_out.e-s002'
+    requirement = "The Adaptivity system shall support high-order SCALAR variables"
```
Comment (cticenhour) [inline test/tests/adaptivity/scalar/tests:8]:
```suggestion
    requirement = "The system shall support adaptivity with high-order SCALAR variables"
```

This fits the requirement standard we've chosen a bit better.

**Review by cticenhour — CHANGES_REQUESTED** [review]

**Review by lindsayad — DISMISSED** [review]

File: test/tests/adaptivity/scalar/tests (hunk 1 above)
Comment (lindsayad) [reply test/tests/adaptivity/scalar/tests:8]:
```suggestion
    requirement = "The system shall support adaptivity with high-order scalar variables"
```
Slight tweak from @cticenhour

Comment (cticenhour) [reply test/tests/adaptivity/scalar/tests:8]: Ah yeah - that should be lowercase.

**Review by cticenhour — APPROVED** [review]
Thanks all!

**Pull request merged**